invert.c

```diff
@@ -0,0 +1,62 @@
+#include <stdio.h>
+#include <limits.h>
+
+char *negative_number(int, char *);
+void breakup_bits_into_groups(const char *);
+int invert_last_number(int number, int bits);
+
+int main(void)
+{
+   char bin_str_array[CHAR_BIT * sizeof(int) + 1];
+   int number;
+   puts("Enter integers and see them in binary.");
+   puts("Non-numeric input terminates program.");
+
+   while(scanf("%d", &number) == 1)
+   {
+      negative_number(number, bin_str_array);
+      printf("%d is \n", number);
+      breakup_bits_into_groups(bin_str_array);
+      putchar('\n');
+      number = invert_last_number(number, 4);
+      printf("Inverting the last 4 bits gives\n");
+      breakup_bits_into_groups(negative_number(number, bin_str_array));
+      putchar('\n');
+   }
+   puts("Bye!");
+
+   return 0;
+}
+
+char *negative_number(int negative_number, char *ps)
```

Comment:
I think this function name should be improved. When I looked at the name of the function and the arguments I had no idea what this function did.